cmd/up/up.go

```diff
@@ -972,22 +972,43 @@ func terminateChildProcess(parent int, pList []ps.Process) {
 }
 
 func terminateProcess(pid int) error {
+	oktetoLog.Debugf("terminating process: %s", pid)
+
 	p, err := os.FindProcess(pid)
 	if err != nil {
 		oktetoLog.Debugf("error getting process %s: %v", pid, err)
 		return err
 	}
-	if err := p.Signal(syscall.SIGTERM); err != nil {
-		if errors.Is(err, os.ErrProcessDone) {
-			return nil
-		}
-		oktetoLog.Debugf("error terminating process %s: %v", p.Pid, err)
+
+	pgid, err := syscall.Getpgid(pid)
+	if err != nil {
+		oktetoLog.Debugf("error getting process group id %s: %v", pid, err)
 		return err
 	}
-	if _, err := p.Wait(); err != nil {
-		oktetoLog.Debugf("error waiting for process to exit %s: %v", p.Pid, err)
-		return err
+
+	if pgid == pid {
+		// if the pid equals its process group id (pgid), it's a session leader process such as bash
```

Comment:
🫠 wow! nice research..... i am wondering if using bash is decided by us i recall? i mean, the user just uses "yarn start" but this is executed with bash....
- what would happen in windows without bash?
- have we tried to run the yarn start with sh?

Reply:
- In the example reported, `bash` is the command in `okteto.yml`
- In Windows it's easier because it's directly sending a `SIGKILL`
- I've tried with `yarn start` inside `sh` and works too :)